bootstrap/cmd/bootstrap/app/kfctlClient.go

```diff
@@ -75,9 +79,45 @@ func NewKfctlClient(instance string) (KfctlService, error) {
 }
 
 // CreateDeployment issues a CreateDeployment to the requested backend
-func (c *KfctlClient) CreateDeployment(ctx context.Context, req kfdefs.KfDef) (*kfdefs.KfDef, error) {
+func (c *KfctlClient) CreateDeployment(ctx context.Context, req C2DRequest) (*kfdefs.KfDef, error) {
+	configFileBytes, err := configconverters.LoadFileFromURI(req.ConfigFile)
+	if err != nil {
+		deployReqCounter.WithLabelValues("INTERNAL").Inc()
+		return nil, err
+	}
+	// appdir will be reset on kfctl server.
+	kfconfigIns, err := configconverters.KfdefByteToKfConfig(configFileBytes, "/opt/bootstrap")
+	if err != nil {
+		deployReqCounter.WithLabelValues("INTERNAL").Inc()
+		log.Errorf("Failed to convert kfdef-byte to kfconfig: %v", err)
+		return nil, err
+	}
+	kfconfigIns.Spec.ConfigFileName = filepath.Base(req.ConfigFile)
+
+	// Fill in user input
+	kfconfigIns.Name = req.Name
+	// Not passing a pointer interface is a common cause of deserialization problems
+	pluginSpec := &gcp.GcpPluginSpec{}
+
+	err = kfconfigIns.GetPluginSpec(gcp.GcpPluginName, pluginSpec)
+	if err != nil {
+		deployReqCounter.WithLabelValues("INTERNAL").Inc()
```

Comment:
What does INTERNAL mean here?

Reply:
INTERNAL mark the error as true error, instead of false positives like invalid input fields.